src/pure_pursuit.cpp

```diff
@@ -103,36 +105,38 @@ double PurePursuit::path_length()
 {
 	double totalLength = 0;
 
-	for (uint32_t i = 0; i < m_robot_path.size() - 1; i++)
-	{
+	for ( int i = 0; i < ( m_robot_path.size() - 1 ); i++ )  // will use current point (i) and next point (i+1)
+	{            																						// until i = the second to last point in vector
 		totalLength += distanceFormula( m_robot_path.at( i ), m_robot_path.at( i + 1 ) );
 	}
 
-	return totalLength;
+		return totalLength;
 }
 
 // function to scale values in one range to values in another range -  proportional
 // scaling.. this is to find z value
-double PurePursuit::pscale(const double& x, const double& in_min, const double& in_max, const double& out_min, const double& out_max)
+double PurePursuit::pscale( const double& x, const double& in_min, const double& in_max,
+	const double& out_min, const double& out_max )
 {
 	return ( x - in_min ) * ( out_max - out_min ) / ( in_max - in_min ) + out_min;
 }
 
-void PurePursuit::reset_path(const Path& robot_path)
+void PurePursuit::reset_path( const Path& robot_path ) 
```

Comment:
please remove the strange spacing in function prototypes too

Reply:
^^^